governance/pyth_staking_sdk/src/pyth-staking-client.ts

```diff
@@ -0,0 +1,204 @@
+  getStakeAccountCustodyAddress,
+} from "./pdas";
+import type { GlobalConfig, PoolConfig, PoolDataAccount } from "./types";
+import {
+  StakeAccountPositions,
+  StakeAccountPositionsAnchor,
+} from "./staking/accounts";
+import type { IntegrityPool } from "../types/integrity_pool";
+import {
+  type Account,
+  createTransferInstruction,
+  getAccount,
+  getAssociatedTokenAddress,
+} from "@solana/spl-token";
+import {
+  sendTransactions,
+  TransactionBuilder,
+} from "@pythnetwork/solana-utils";
+import { convertBigIntToBN, convertBNToBigInt } from "./utils";
+import type { AnchorWallet } from "@solana/wallet-adapter-react";
+
+export type PythStakingClientConfig = {
+  connection: Connection;
+  wallet: AnchorWallet;
+};
+
+export class PythStakingClient {
+  connection: Connection;
+  wallet: AnchorWallet;
+  provider: AnchorProvider;
+  stakingProgram: Program<Staking>;
+  integrityPoolProgram: Program<IntegrityPool>;
+
+  constructor(config: PythStakingClientConfig) {
+    this.connection = config.connection;
+    this.wallet = config.wallet;
+    this.provider = new AnchorProvider(this.connection, this.wallet, {
+      skipPreflight: true,
+    });
+    this.stakingProgram = new Program(StakingIdl as Staking, this.provider);
+    this.integrityPoolProgram = new Program(
+      IntegrityPoolIdl as IntegrityPool,
+      this.provider
+    );
+  }
+
+  async setGlobalConfig(config: GlobalConfig) {
+    const globalConfigAnchor = convertBigIntToBN(config);
+    return this.stakingProgram.methods.initConfig(globalConfigAnchor).rpc();
+  }
+
+  async getGlobalConfig(): Promise<GlobalConfig> {
+    const globalConfigAnchor =
+      await this.stakingProgram.account.globalConfig.fetch(
+        getConfigAddress()[0]
+      );
+    return convertBNToBigInt(globalConfigAnchor);
+  }
+
+  /** Gets a users stake accounts */
+  public async getStakeAccountPositions(
+    user: PublicKey
+  ): Promise<StakeAccountPositions[]> {
+    const res =
+      await this.stakingProgram.provider.connection.getProgramAccounts(
+        this.stakingProgram.programId,
+        {
+          encoding: "base64",
+          filters: [
+            {
+              memcmp: this.stakingProgram.coder.accounts.memcmp("positionData"),
+            },
+            {
+              memcmp: {
+                offset: 8,
+                bytes: user.toBase58(),
+              },
+            },
+          ],
+        }
+      );
+    return res.map((account) => {
+      const stakeAccountPositionsAnchor = new StakeAccountPositionsAnchor(
+        account.pubkey,
+        account.account.data,
+        this.stakingProgram.idl
+      );
+      return stakeAccountPositionsAnchor.toStakeAccountPositions();
+    });
+  }
+
+  public async getStakeAccountCustody(
+    stakeAccountPositions: PublicKey
+  ): Promise<Account> {
+    return getAccount(
+      this.connection,
+      getStakeAccountCustodyAddress(stakeAccountPositions)
+    );
+  }
+
+  public async initializePool({
```

Comment:
Not entirely convinced about having these things in this sdk since they're not called by users of the app

Reply:
i added them for tests but i agree with your point. But let's deal with them later, i don't want to remove the tests right now.